app/forms/school_form.rb

```diff
@@ -0,0 +1,29 @@
+class SchoolForm
+  include ActiveModel::Model
+
+  attr_accessor :urn, :service
+
+  validates :urn, presence: true
+  validates :service, presence: true, inclusion: { in: %i[placements claims] }
+  validate :school_already_onboarded?
+
+  def onboard
+    return false unless valid?
+
+    school.update!(service => true)
+  end
+
+  def school
+    @school ||= School.find_by(urn:)
+  end
+
+  private
+
+  def school_already_onboarded?
+    if school.blank?
+      errors.add(:urn, :blank)
```

Comment:
I also wonder if `:blank` is the correct message to display here? 🤔

Although it seems to be an edge case, because it should (in theory) only happen if the user fiddles with the URL parameter themselves. So perhaps it's fine to leave this as is.

Reply:
See above.